react-app/src/view/UserAuth/SignUp.jsx

```diff
@@ -0,0 +1,188 @@
+import React from "react";
+import {
+  Typography,
+  Container,
+  Box,
+  TextField,
+  Button,
+  Grid,
+} from "@mui/material";
+import { Link, useNavigate } from "react-router-dom";
+import axiosInstance from "../../axios";
+import { base } from "../../api";
+import { useState } from "react";
+import { loginSchema, validateUser } from "../../Validations/Validations";
+import { ToastContainer, toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+
+export default function SignUp() {
+  // const [newUser,setNewUser] = useState([]);
+  const navigate = useNavigate();
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  const signup = async () => {
+    try {
+      const response = await axiosInstance({
+        method: "post",
+        url: base + "/users",
+        data: {
+          user: {
+            first_name: firstName,
+            last_name: lastName,
+            email: email,
+            password: password,
+          },
+        },
+      });
+
+      console.log(response.data.data.attributes);
+      // dispatch(signUp(response.data.data.attributes));
+      // dispatch(setToken(response.data.data.attributes.token));
+      //navigate("/questions");
+    } catch (error) {
+      console.log("error Signing Up");
+    }
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    try {
+      const user = {
+        first_name: firstName,
+        last_name: lastName,
+        email: email,
+        password: password,
+      };
+      console.log(user);
+      const isValid = await validateUser(user, loginSchema);
+
+      console.log("isValid", isValid);
+
+      // TODO: display validation errors
+      if (!isValid)
+        return toast.error("wrong email or password!", {
+          position: "top-center",
+          autoClose: 5000,
+          hideProgressBar: false,
+          closeOnClick: true,
+          pauseOnHover: true,
+          draggable: true,
+          progress: undefined,
+        });
+
+      signup();
+      navigate("/questions");
```

Comment:
`await` signup before navigating to another page, because you want to make sure it succeeds before navigating.